pkg/download/addons/with_pool.go

```diff
@@ -0,0 +1,121 @@
+package addons
+
+import (
+	"context"
+	"io"
+
+	"github.com/gomods/athens/pkg/download"
+	"github.com/gomods/athens/pkg/errors"
+	"github.com/gomods/athens/pkg/storage"
+)
+
+type withpool struct {
+	dp download.Protocol
+	ch chan func()
```

Comment:
can you rethink the name? what is ch doing signaling done or accepting work?
from looking at the code below I can read that, but it would be nice to know this straight away

Reply:
since this is a worker pool, I think `jobCh` makes sense. Would that be ok? I can also add some internal docs for it :)